src/shogun/machine/gp/SparseInference.cpp

```diff
@@ -99,16 +97,17 @@ SparseInference::SparseInference(std::shared_ptr<Kernel> kern, std::shared_ptr<F
 
 void SparseInference::init()
 {
-	SG_ADD(&m_inducing_features, "inducing_features", "inducing features",
-			ParameterProperties::HYPER | ParameterProperties::GRADIENT);
+	SG_ADD(
+	    &m_inducing_features, "inducing_features", "inducing features",
+	    ParameterProperties::HYPER | ParameterProperties::GRADIENT);
 	SG_ADD(&m_log_ind_noise, "log_inducing_noise", "noise about inducing potins in log domain",
 		ParameterProperties::HYPER | ParameterProperties::GRADIENT);
 	SG_ADD(&m_mu, "mu", "mean vector of the approximation to the posterior");
 	SG_ADD(&m_Sigma, "Sigma", "covariance matrix of the approximation to the posterior");
 	SG_ADD(&m_ktrtr_diag, "ktrtr_diag", "diagonal elements of kernel matrix m_ktrtr");
 
 	m_log_ind_noise = std::log(1e-10);
-	m_inducing_features=SGMatrix<float64_t>();
+	m_inducing_features = nullptr;
```

Comment:
I think the shared_ptr default constructor takes care of this, no need to have this line